src/plugins/interactive_setup/public/cluster_configuration_form.tsx

```diff
@@ -0,0 +1,332 @@
+import type { ValidationErrors } from './use_form';
+import { useForm } from './use_form';
+import { useHtmlId } from './use_html_id';
+import { useHttp } from './use_http';
+
+export interface ClusterConfigurationFormValues {
+  username: string;
+  password: string;
+  caCert: string;
+}
+
+export interface ClusterConfigurationFormProps {
+  host: string;
+  authRequired: boolean;
+  certificateChain?: Certificate[];
+  defaultValues?: ClusterConfigurationFormValues;
+  onCancel?(): void;
+  onSuccess?(): void;
+}
+
+export const ClusterConfigurationForm: FunctionComponent<ClusterConfigurationFormProps> = ({
+  host,
+  authRequired,
+  certificateChain,
+  defaultValues = {
+    username: 'kibana_system',
+    password: '',
+    caCert: '',
+  },
+  onCancel,
+  onSuccess,
+}) => {
+  const http = useHttp();
+
+  const [form, eventHandlers] = useForm({
+    defaultValues,
+    validate: async (values) => {
+      const errors: ValidationErrors<ClusterConfigurationFormValues> = {};
+
+      if (authRequired) {
+        if (!values.username) {
+          errors.username = i18n.translate(
+            'interactiveSetup.clusterConfigurationForm.usernameRequiredError',
+            {
+              defaultMessage: 'Enter a username.',
+            }
+          );
+        } else if (values.username === 'elastic') {
+          errors.username = i18n.translate(
+            'interactiveSetup.clusterConfigurationForm.usernameReservedError',
+            {
+              defaultMessage: "User 'elastic' can't be used as Kibana system user.",
+            }
+          );
+        }
+
+        if (!values.password) {
+          errors.password = i18n.translate(
+            'interactiveSetup.clusterConfigurationForm.passwordRequiredError',
+            {
+              defaultMessage: `Enter a password.`,
+            }
+          );
+        }
+      }
+
+      if (certificateChain && !values.caCert) {
+        errors.caCert = i18n.translate(
+          'interactiveSetup.clusterConfigurationForm.caCertConfirmationRequiredError',
+          {
+            defaultMessage: 'Confirm that you recognize and trust this certificate.',
+          }
+        );
+      }
+
+      return errors;
+    },
+    onSubmit: async (values) => {
+      try {
+        await http.post('/internal/interactive_setup/configure', {
+          body: JSON.stringify({
+            hosts: [host],
+            username: values.username,
+            password: values.password,
+            caCert: values.caCert,
+          }),
+        });
+        onSuccess?.();
+      } catch (error) {
+        if (error.body?.message.includes('self signed certificate in certificate chain')) {
+          form.setError('caCert', 'Certificate is not valid for this cluster.');
+        } else if (error.body?.message.includes('unable to authenticate user')) {
+          form.setError('password', 'Invalid username or password.');
+        } else {
+          throw error;
+        }
+      }
```

Comment:
__nit:__ maybe we should parse this on the server side and return a proper error type in metadata as we discussed previously?

Reply:
Agreed, I'll remove these for now. We will improve error messages and guidance in a later PR.